### PR TITLE
[ML] add new trained model deployment cache clear API

### PR DESCRIPTION
**Pull request by benwtrent**
This adds a new `_ml/trained_models/<model_id>/deployment/cache/_clear` API. This will clear the inference cache on every node where the model is allocated.

### CODE REVIEW

**Comment by droberts195** [comment]
This has quite a major merge clash with #88901.  I'll take it over once #88901 is merged.  elastic/ml-cpp#2376 slightly changed the results structure, so if this PR were merged first then there would be two PRs that require coordination between the C++ and Java side.  But if #88901 is merged first and then this PR updated to match then there'll only be one.

**Comment by sethmlarson** [comment]
Looks good from an API perspective! :+1:

File: docs/reference/ml/trained-models/apis/clear-trained-model-deployment-cache.asciidoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+preview::[]
+
+[[clear-trained-model-deployment-cache-request]]
+== {api-request-title}
+
+`POST _ml/trained_models/<model_id>/deployment/cache/_clear`
+
+[[clear-trained-model-deployment-cache-prereq]]
+== {api-prereq-title}
+Requires the `manage_ml` cluster privilege. This privilege is included in the
```
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/clear-trained-model-deployment-cache.asciidoc:20]:
```suggestion
== {api-prereq-title}

Requires the `manage_ml` cluster privilege. This privilege is included in the
```

**Review by szabosteve — APPROVED** [review]
One suggestion about formatting, otherwise docs LGTM.

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by droberts195 — APPROVED** [review]
LGTM2

**Pull request merged**